editor/src/document/document_message_handler.rs

```diff
@@ -219,16 +235,32 @@ impl MessageHandler<DocumentsMessage, &InputPreprocessor> for DocumentsMessageHa
 			NewDocument => {
 				let name = self.generate_new_document_name();
 				let new_document = DocumentMessageHandler::with_name(name, ipp);
-				self.load_document(new_document, responses);
+				self.load_document(new_document, generate_uuid(), false, responses);
 			}
 			OpenDocument => {
 				responses.push_back(FrontendMessage::OpenDocumentBrowse.into());
 			}
-			OpenDocumentFile(name, serialized_contents) => {
-				let document = DocumentMessageHandler::with_name_and_content(name, serialized_contents, ipp);
+			OpenDocumentFile(document_name, document) => responses.push_back(
```

Comment:
For readability and consistency, I think it's better to use a `{ }` block instead of directly `=> responses.push_back(...),`